prometheus_aws_guardduty_exporter/collector.py

```diff
@@ -23,30 +43,28 @@ def collect(self):
             labels=["region"])
 
         # Customize the boto client config
-        botoConfig = botocore.client.Config(connect_timeout=2, read_timeout=10, retries={"max_attempts": 2})
+        self.botoConfig = botocore.client.Config(connect_timeout=2, read_timeout=10, retries={"max_attempts": 2})
 
+        threads = []
         for region in self.regions:
-            try:
-                # Scrape GuardDuty statistics from the region
-                regionStats = self._collectMetricsByRegion(region, botoConfig)
+            # We start one thread per region.
+            process = threading.Thread(target=self.scrape_metric_in_region, args=[currentFindingsMetric, region])
+            threads.append(process)
 
-                # Add the findings to the exporterd metric
-                for severity, count in regionStats.items():
-                    currentFindingsMetric.add_metric(value=count, labels=[region, severity])
-            except Exception as error:
-                logging.getLogger().error(f"Unable to scrape GuardDuty statistics from {region} because of error: {str(error)}")
+        for process in threads:
+            process.start()
```

Comment:
You can avoid this iteration, moving `process.start()` right before `threads.append(process)`